src/main/java/com/google/api/codegen/config/FieldConfig.java

```diff
@@ -212,10 +212,14 @@ public FieldConfig withResourceNameConfig(ResourceNameConfig resourceNameConfig)
   public boolean requiresParamTransformation() {
     return getResourceNameConfig() != null
         && getMessageResourceNameConfig() != null
-        && getMessageResourceNameConfig().getResourceNameType() != ResourceNameType.ANY
         && !getResourceNameConfig().equals(getMessageResourceNameConfig());
   }
 
+  public boolean requiresAnyParamTransformation() {
```

Comment:
Honestly, I don't know I keep wondering how much unexpected complexity is in those resource names. 

I feel like forcing to string for equality check is somewhat against the resource name philosophy (as they exist to make things more static). At the same time that sill may be a reasonable approach here, to emphasize that resource names are only 1) client side 2) wrapper classes with utility methods and 3) the actual entity a user is working with is still a string.